packages/expo-document-picker/ios/DocumentPickerModule.swift

```diff
@@ -0,0 +1,203 @@
+import ExpoModulesCore
+import UIKit
+import MobileCoreServices
+import Foundation
+
+/**
+ Helper struct storing single picking operation context variables that have their own non-sharable state.
+ */
+struct PickingContext {
+  let promise: Promise
+  let options: DocumentPickerOptions
+  let documentPickerHandler: DocumentPickerHandler
+}
+
+public class DocumentPickerModule: Module, DocumentPickingResultHandler {
+  public func definition() -> ModuleDefinition {
+    Name("ExpoDocumentPicker")
+
+    AsyncFunction("getDocumentAsync") { (options: DocumentPickerOptions, promise: Promise) -> Void in
+      // TODO: do we need this?
```

Comment:
You would check your picking context is not nil indicating picking is in progress.